File: backend/app/schemas/subscription.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class SubscriptionBase(BaseModel):
    user_id: int
    mollie_id: str
    customer_id: str
    package_id: int
    package_type: str
    project_limit: Optional[int]
    status: str
    amount: float
    interval: str
    start_date: datetime
    end_date: Optional[datetime]
    last_payment_date: Optional[datetime]
    next_payment_date: Optional[datetime]

class SubscriptionCreate(SubscriptionBase):
    pass

class SubscriptionUpdate(BaseModel):
    status: Optional[str] = None
    project_limit: Optional[int] = None
    end_date: Optional[datetime] = None

class SubscriptionResponse(SubscriptionBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/subscription.ts
import { User } from './api';

export type SubscriptionTier = 'trial' | 'team' | 'enterprise';
export type SubscriptionStatus = 'active' | 'cancelled' | 'expired';

export interface SubscriptionPlan {
  id: number;
  name: string;
  tier: SubscriptionTier;
  price: number;
  interval: string;
  projectLimit: number;
  features: string[];
  buttonText: string;
  isActive: boolean;
}

export interface Subscription {
  id: number;
  userId: number;
  mollieId: string;
  customerId: string;
  packageId: number;
  packageType: SubscriptionTier;
  projectLimit: number;
  status: SubscriptionStatus;
  amount: number;
  interval: string;
  startDate: string;
  endDate?: string;
  lastPaymentDate?: string;
  nextPaymentDate?: string;
  createdAt: string;
}

export interface SubscriptionDetails extends Subscription {
  user: User;
  plan: SubscriptionPlan;
}

export interface CancellationRequest {
  subscriptionId: number;
  reason?: string;
  feedback?: string;
}

export interface SubscriptionUpdateRequest {
  status?: SubscriptionStatus;
  projectLimit?: number;
  endDate?: string;
}
